tests/10-issue.js

```diff
@@ -26,95 +26,332 @@ describe('BitstringStatusList Credentials (Issue "statusPurpose: revocation")',
+                  async function() {
+                    this.test.link = 'https://www.w3.org/TR/vc-bitstring-status-list/#:~:text=statusMessage%20MAY%20be%20present%20if%20statusSize%20is%201%2C%20and%20MUST%20be%20present%20if%20statusSize%20is%20greater%20than%201';
+                    issuedVc.credentialStatus.should.have.own.property(
+                      'statusMessage');
+                  });
+                }
+                it('If the statusMessage array is present, each ' +
+                  'element MUST contain the two properties "status" ' +
+                  'and "message".',
+                async function() {
+                  this.test.link = 'https://www.w3.org/TR/vc-bitstring-status-list/#:~:text=If%20the%20statusMessage%20array%20is%20present%2C%20each%20element%20MUST%20contain%20the%20two%20properties%20described%20below%2C%20and%20MAY%20contain%20additional%20properties.';
+                  // TODO this needs more testing:
+                  // a string representing the hexadecimal
+                  // value of the status prefixed with 0x
+                  const statusMessage = issuedVc.credentialStatus.statusMessage;
+                  statusMessage.should.each.have.property(
+                    'status').that.is.a('string');
+                  statusMessage.should.each.have.property(
+                    'message').that.is.a('string');
+                });
+              } else {
+                it.skip('If statusSize is not present as a property ' +
+                  'of the credentialStatus, then statusSize MUST ' +
+                  'be processed as 1.',
+                async function() {
+                  this.test.link = 'https://www.w3.org/TR/vc-bitstring-status-list/#:~:text=If%20statusSize%20is%20not%20present%20as%20a%20property%20of%20the%20credentialStatus%2C%20then%20statusSize%20MUST%20be%20processed%20as%201.';
+                }
+                );
+              }
+            });
+          it('An implementer MAY include the statusReference property.',
+            async function() {
+              this.test.link = 'https://www.w3.org/TR/vc-bitstring-status-list/#:~:text=An%20implementer%20MAY%20include%20the%20statusReference%20property.';
+              if(!!issuedVc.credentialStatus.statusReference) {
+                it('If present, its value MUST be a URL or an array of URLs ' +
+                    '[URL] which dereference(s) to material related to the ' +
+                    'status.',
+                async function() {
+                  this.test.link = 'https://www.w3.org/TR/vc-bitstring-status-list/#:~:text=If%20present%2C%20its%20value%20MUST%20be%20a%20URL%20or%20an%20array%20of%20URLs%20%5BURL%5D%20which%20dereference%20to%20material%20related%20to%20the%20status';
+                  const statusReferenceType = typeof (
+                    issuedVc.credentialStatus.statusReference);
+                  statusReferenceType.should.be.oneOf(['string', 'array'],
```

Comment:
Same problem as before: `typeof` won't ever return `array`.

Reply:
changed to object and added a more explicit test on the object to see if its an array